lib/pbench/server/api/resources/query_apis/datasets_owner.py

```diff
@@ -0,0 +1,101 @@
+from logging import Logger
+from typing import Iterator
+
+from pbench.server import JSONOBJECT, OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    ApiAuthorizationType,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.api.resources.query_apis import ApiContext, ElasticBulkBase
+from pbench.server.database.models.datasets import Dataset
+
+
+class DatasetOwner(ElasticBulkBase):
+    """
+    Change the "owner" authorization of a Pbench dataset which is either public or
+    needs change in ownership, by modifying the "authorization": {"owner": owner_id}
+    subdocument of each Elasticsearch document associated with the specified dataset.
```

Comment:
Maybe: "Change the owner of a Pbench dataset by modifying the "owner" in the Dataset SQL table and in each Elasticsearch document "authorization" sub-document associated with the dataset." (??)